CODINGSTYLE.md

```diff
@@ -0,0 +1,335 @@
+# Coding Style Guide
+
+The Scala.js project has pretty drastic rules in terms of coding style.
+This document tries to document them as much as possible.
+
+A *few* of these rules are checked automatically using Scalastyle, but most of them are too complex to teach to an automated tool.
+
+In some cases, [@sjrd](https://github.com/sjrd) will just say something else on a code review.
+His opinion on a direct piece of code always takes precedence over this document.
+Feel free to point out the conflict, though: the document can be improved.
```

Comment:
@sjrd wrote scala.js so, as it happens, his code style and scala.js' coincide, but scala.js supersedes a single person, it's a project with a core team and the document should be about the core team's code style IMHO

tl;dr code style is @sjrd 's, present it as scala.js core team's style to show a united front, you're all set if you want to change something in the future and at the same time you're not giving the impression of a BDFL